vhdl_parser/src/tokenstream.rs

```diff
@@ -102,6 +102,17 @@ impl TokenStream {
         Ok(self.pop_if_kind(kind)?.is_some())
     }
 
+    pub fn skip_to(self: &mut Self, kinds: &[Kind]) -> ParseResult<()> {
```

Comment:
better make this an `Fn` that returns true or false depending on if the kind is correct or not. Vec::contains is O(n)

Reply:
I guess I could adapt it to have a closure as a second parameter, to feed the match statement from each call site into it without duplicating it for each distinct list of recovery tokens.

From a performance point of view, it's not clear to me that a O(n) linear search is actually slower than jump tables.
We would need multiple jump tables, each one with >100 entries to match on all possible values of the token kind.
The list to search on the other hand would be at most around 20 elements large and I think the typical case will be around 1-2 Elements (e.g. recovering to a semicolon).

Reply:
I think it gets implementet as a shift and bitwise and for enums less than 128 kinds.